### PR TITLE
[PT-5597] Fixing test for catalog class

### PR DESCRIPTION
**Pull request by andher1802**
See changelog.

Checklist:
* [x] Test coverage
* [x] Version bump
* [x] Changelog update

### CODE REVIEW

File: tests/test_catalog.py [hunk 1]
```diff
@@ -180,50 +228,63 @@ def test_should_download_available_quicklooks(self, requests_mock: req_mock.Mock
...
+        [
+            {"usage_type": ["DATA"]},
+            {"usage_type": ["ANALYTICS"]},
+            {"usage_type": ["DATA", "ANALYTICS"]},
+            {},
+        ],
     )
-
-
-def test_construct_search_parameters_fc_multiple_features_raises(catalog_mock):
```
Comment (andher1802) [inline tests/test_catalog.py:199]: This is covered in test_utils and not related to catalog

File: up42/catalog.py [hunk 2]
```diff
@@ -91,6 +112,8 @@ class Catalog(CatalogBase):
     [CatalogBase](catalogbase-reference.md) class.
     """
 
+    session = base.Session()
```
Comment (javidq) [inline up42/catalog.py:115]: Duplicate, not needed

**Review by javidq — CHANGES_REQUESTED** [review]

File: tests/test_catalog.py [hunk 3]
```diff
@@ -1,30 +1,40 @@
...
-    },
+    "limit": 10,
 }
+FEATURE = {
+    "type": "Feature",
+    "properties": {"some": "data"},
+    "geometry": {"type": "Point", "coordinates": (1.0, 2.0)},
+}
+POINT_BBOX = (1.0, 2.0, 1.0, 2.0)
+Geometries = catalog.Geometries
```
[the remark this answers is not in the document]
Comment (andher1802) [reply tests/test_catalog.py:37]: Using directly the catalog.Geometry type triggers a pylint warning. Variable not allowed in type expressionPylance[reportInvalidTypeForm](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportInvalidTypeForm). We can add an ignore, remove the typing or add the typing alias as current.

File: tests/test_catalog.py [hunk 4]
```diff
@@ -43,16 +53,64 @@ def set_status_raising_session():
...
+        assert isinstance(order_obj, order.Order)
+        assert order_obj.track_status(report_time=0.1) == "FULFILLED"
 
 
 class TestCatalog:
     host = "oneatlas"
     catalog = catalog.Catalog(auth=mock.MagicMock(), workspace_id=constants.WORKSPACE_ID)
 
+    @pytest.fixture(params=["output_dir", "no_output_dir"])
+    def output_directory(self, request, tmp_path) -> Optional[str]:
```
[the remark this answers is not in the document]
Comment (andher1802) [reply tests/test_catalog.py:111]: it is Generator[Path, None, None], a Path object.

File: tests/test_catalog.py [hunk 5]
```diff
@@ -272,9 +278,10 @@ def test_should_download_available_quicklooks(self, requests_mock: req_mock.Mock
         out_paths = self.catalog_obj.download_quicklooks(
             image_ids=[image_id, missing_image_id],
             collection=PHR,
-            output_directory=tmp_path,
+            output_directory=output_directory,
         )
-        assert out_paths == [str(tmp_path / f"quicklook_{image_id}.jpg")]
+        download_folder = output_directory if output_directory else pathlib.Path.cwd() / "catalog"
```
Comment (javidq) [inline tests/test_catalog.py:283]: `output_directory or pathlib.Path.cwd() / "catalog"`

File: CHANGELOG.md [hunk 6]
```diff
@@ -29,6 +29,10 @@ You can check your current version with the following command:
     ```
 
 For more information, see [UP42 Python package description](https://pypi.org/project/up42-py/).
+## 2.1.0a10
+
+**Oct 7, 2024**
```
Comment (javidq) [inline CHANGELOG.md:34]: To be updated

**Review by javidq — CHANGES_REQUESTED** [review]

**Review by javidq — APPROVED** [review]

**Pull request merged**